semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/linking/filecompat/FileNameCompat.scala

```diff
@@ -18,7 +18,8 @@ class FileNameCompat(cpg: Cpg) extends CpgPass(cpg) {
     val dstGraph = DiffGraph.newBuilder
 
     def updateDefaultFileName(node: nodes.StoredNode with nodes.HasFilename): Unit = {
-      if (node.filename == null) {
+      // When creating nodes via NewNode classes, filename is "", not null.
+      if (node.filename == null || node.filename == "") {
```

Comment:
This is the one change that may be problematic, so, if we see trouble down the line, this may need to be reverted.